File: Difficulty/Medium/[96] Unique Binary Search Trees/solution.py
class Solution:
    def numTrees(self, n: int) -> int:
        if n <= 0:
            return 0
        h_n = 1
        for i in range(1, n + 1):
            h_n = (4 * i - 2) * h_n // (i + 1)
        return h_n
            

File: Difficulty/Medium/[96] Unique Binary Search Trees/solution.cpp
/*
 * @lc app=leetcode id=96 lang=cpp
 *
 * [96] Unique Binary Search Trees
 */
class Solution {
public:
    int numTrees(int n) {
        if (n <= 0) return 0;
        long h_n = 1;
        for (int i = 1; i <= n; ++i)
            h_n = (4 * i - 2) * h_n / (i + 1);
        return h_n;
    }
};

